arch/arm64/boot/dts/qcom/msm8953.dtsi

```diff
@@ -633,6 +633,24 @@
 		reg = <0 0 0 0>;
 	};
 
+	mpm: interrupt-controller@601d4 {
+		compatible = "qcom,mpm";
+		qcom,rpm-msg-ram = <&apss_mpm>;
+		interrupts = <GIC_SPI 171 IRQ_TYPE_EDGE_RISING>;
+		mboxes = <&apcs 1>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
+		#power-domain-cells = <0>;
+		interrupt-parent = <&intc>;
+		qcom,mpm-pin-count = <96>;
+		qcom,mpm-pin-map = <2 184>,  /* tsens_upper_lower_int */
```

Comment:
It seems it is not matching with this https://git.codelinaro.org/clo/la/kernel/msm-4.9/-/blob/LA.UM.9.6.2.r1-04800-89xx.0/drivers/irqchip/qcom/mpm-8953.c#L16

Reply:
This offset IS confusing, but this is due to the the different mpm drivers in mainline and downstream kernel. The downstream kernel routes every interrupt through the mpm.

The irq numbers in mpm-pin-map should differ by 32 due to this, I think my template for this was 
```
commit 09896da07315cce07b019ab00750c8a57e1b53a3
Author: Konrad Dybcio <konradybcio@kernel.org>
Date:   Fri Dec 15 01:01:09 2023 +0100

    arm64: dts: qcom: msm8996: Hook up MPM
    
    Wire up MPM and the interrupts it provides.
    
    Signed-off-by: Konrad Dybcio <konrad.dybcio@linaro.org>
    Link: https://lore.kernel.org/r/20231215-topic-mpm_dt-v1-2-c6636fc75ce3@linaro.org
    Signed-off-by: Bjorn Andersson <andersson@kernel.org>

```
This has since then been reverted 
```
commit 4f423c4cbe26d79d8974936eb01e0d6574c5d2ac
Author: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date:   Wed Feb 21 01:07:21 2024 +0200

    Revert "arm64: dts: qcom: msm8996: Hook up MPM"
    
    Commit 09896da07315 ("arm64: dts: qcom: msm8996: Hook up MPM") has
    hooked up the MPM irq chip on the MSM8996 platform. However this causes
    my Dragonboard 820c crash during bootup (usually when probing IOMMUs).
    Revert the offending commit for now. Quick debug shows that making
    tlmm's wakeup-parent point to the MPM is enough to trigger the crash.
    
    Fixes: 09896da07315 ("arm64: dts: qcom: msm8996: Hook up MPM")
    Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
    Link: https://lore.kernel.org/r/20240221-msm8996-revert-mpm-v1-1-cdca9e30c9b4@linaro.org
    Signed-off-by: Bjorn Andersson <andersson@kernel.org>

```
I think I had the mentioned problem, too, (crash on boot when using mpm as tlmm's wakeup-parent) but solved it (as in avoiding the crash, wakeup is still not working). I will look into it but first have to rebase these to the newest branch, I havent't booted my FP3 into debian for a long time.

Reply:
I tried to find an example (i.e. an android kernel for msm8996 where the wakegic is used in the devicetree) but couldn't find one, so I tried to use the data from mpm-8953.c in my devicetree without shifting:
```
		qcom,mpm-pin-map = <2 216>,  /* tsens_upper_lower_int */
				   <37 252>, /* qmp_usb3_lfps_rxterm_irq -> ss_phy_irq */
				   <49 168>, /* qusb2phy_dpse_hv -> hs_phy_irq*/
				   <53 104>, /* mdss_irq */
				   <58 168>, /* qusb2phy_dmse_hv -> hs_phy_irq*/
				   <88 222>; /* ee0_krait_hlos_spmi_periph_irq */

```
and with this the kernel hangs when booting. I take this is a hint that the shift is correct.